diff --git a/m4/guile.m4 b/m4/guile.m4
index 696897364..4bc59bf7c 100644
--- a/m4/guile.m4
+++ b/m4/guile.m4
@@ -1,6 +1,7 @@
 ## Autoconf macros for working with Guile.
 ##
 ##   Copyright (C) 1998,2001, 2006, 2010, 2012, 2013, 2014 Free Software Foundation, Inc.
+##   Copyright (C) 2020 Lepton EDA Contributors
 ##
 ## This library is free software; you can redistribute it and/or
 ## modify it under the terms of the GNU Lesser General Public License
@@ -301,6 +302,15 @@ AC_DEFUN([GUILE_PROGS],
   AC_PATH_PROG(GUILD,[guild$_guile_suffix])
   AC_SUBST(GUILD)
 
+
+  # Lepton EDA: detect guile-snarf executable on Fedora 32,
+  #             it is named "guile-snarf2.2" there:
+  #
+  AC_PATH_PROG(GUILE_SNARF,[guile-snarf$_guile_suffix])
+  AC_SUBST(GUILE_SNARF)
+
+
+
   AC_PATH_PROG(GUILE_CONFIG,[guile-config$_guile_suffix])
   AC_SUBST(GUILE_CONFIG)
   if test -n "$GUILD"; then
