diff --git a/attrib/src/gtksheet_2_2.c b/attrib/src/gtksheet_2_2.c
index d0e35e809..028497bbf 100644
--- a/attrib/src/gtksheet_2_2.c
+++ b/attrib/src/gtksheet_2_2.c
@@ -8180,7 +8180,7 @@ gtk_sheet_attach_floating       (GtkSheet *sheet,
                                  GtkWidget *widget,
                                  gint row, gint col)
 {
-  GdkRectangle area;
+  GdkRectangle area = { 0, 0, 0, 0 };
   GtkSheetChild *child;
 
   if(row < 0 || col < 0){
diff --git a/schematic/src/gschem_page_view.c b/schematic/src/gschem_page_view.c
index 7b64e053d..2b44bd7a2 100644
--- a/schematic/src/gschem_page_view.c
+++ b/schematic/src/gschem_page_view.c
@@ -608,10 +608,10 @@ gschem_page_view_invalidate_screen_rect (GschemPageView *view, int left, int top
 void
 gschem_page_view_invalidate_world_rect (GschemPageView *view, int left, int top, int right, int bottom)
 {
-  int screen_bottom;
-  int screen_right;
-  int screen_left;
-  int screen_top;
+  int screen_bottom = 0;
+  int screen_right = 0;
+  int screen_left = 0;
+  int screen_top = 0;
 
   g_return_if_fail (view != NULL);
 
